core/test/test_server.hh

```diff
@@ -61,6 +61,9 @@ class test_server {
                    std::error_code const& err,
                    size_t bytes_transfered);
 
+  std::mutex _m;
+  std::condition_variable _cond;
```

Comment:
The same remark for the condition variable. Just because when it crashes, it is difficult to understand those objects.